subsys/usb/usb_c/usbc_pe_common.c

```diff
@@ -151,14 +149,19 @@ void pe_run(const struct device *dev, const int32_t dpm_request)
 		 */
 		if (dpm_request == REQUEST_PE_HARD_RESET_SEND) {
 			pe_set_state(dev, PE_SNK_HARD_RESET);
-		} else {
-			/* Pass the DPM request along to the state machine */
-			pe->dpm_request = dpm_request;
-			common_dpm_requests(dev);
 		}
 
 		/* Run state machine */
 		smf_run_state(SMF_CTX(pe));
+
+		/*
+		 * Handle any common DPM Requests while in the
+		 * Sink Ready state
+		 */
+		if (pe_get_state(dev) == PE_SNK_READY) {
+			common_dpm_requests(dev);
```

Comment:
Would it make more sense to just call this from the ready state? That would make this function care less about what the PE state is.

Reply:
I was calling common_dpm_requests here because it was common to both Source and Sink but I can just call the function from both ready states as you suggested.